src/sardana/macroserver/test/res/macros/testmacros.py

```diff
@@ -0,0 +1,148 @@
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (91, True)
+        expect_params = ([91], True)
+        macro, _ = self.prepareMacro("testParamsFirstRepeat", *params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+
+class createMacro(Macro):
+
+    def run(self, *args):
+
+        expect_params = (99, [1., 2.])
+        macro, pars = self.createMacro('testParamMacros1', *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (99, 1., 2.)
+        expect_params = (99, [1., 2.])
+        self.runMacro(macro)
+        macro, pars = self.createMacro('testParamMacros1', *params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([92], True)
+        macro, _ = self.createMacro("testParamsFirstRepeat", *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (91, True)
+        expect_params = ([91.], True)
+        macro, _ = self.createMacro("testParamsFirstRepeat", *params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+
+class execMacro(Macro):
+
+    def run(self, *args):
+
+        expect_params = (99, [1., 2.])
+        macro = self.execMacro('testParamMacros1', *expect_params)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (99, 1., 2.)
+        macro = self.execMacro('testParamMacros1', *params)
+        result = macro.data
+        expect_params = (99, [1., 2.])
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([92], True)
+        macro = self.execMacro('testParamsFirstRepeat', *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([99.], True)
+        macro = self.execMacro("testParamsFirstRepeat [99] True")
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([999.], True)
+        macro = self.execMacro("testParamsFirstRepeat", 999, True)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+
+class testParamMacros1(Macro):
```

Comment:
I like the idea of analogy with the parameters examples macros. But we need to keep the macro name and its docsting consitent. For this need, I propose to rename the macro to `pt5_base` and in the docstring use reference to the `pt5` example explaining that we use base python types instead of pool elements e.g. motors.